common/src/main/java/com/wynntils/models/skillpoint/SkillPointModel.java

```diff
@@ -0,0 +1,386 @@
+     * tomes have been unlocked.
+     */
+    public void populateSkillPoints() {
+        McUtils.closeBackgroundContainer();
+
+        Managers.TickScheduler.scheduleNextTick(() -> {
+            assignedSkillPoints.clear();
+            calculateGearSkillPoints();
+            queryTotalAndTomeSkillPoints();
+        });
+    }
+
+    private void calculateGearSkillPoints() {
+        gearSkillPoints.clear();
+        craftedSkillPoints.clear();
+
+        // Cannot combine these loops because of the way the inventory is numbered when a container is open
+        McUtils.inventory().armor.forEach(itemStack -> {
+            Optional<WynnItem> wynnItemOptional = Models.Item.getWynnItem(itemStack);
+            if (wynnItemOptional.isEmpty()) return; // Empty slot
+
+            if (wynnItemOptional.get() instanceof GearItem gear) {
+                gear.getIdentifications().forEach(x -> {
+                    if (x.statType() instanceof SkillStatType skillStat) {
+                        gearSkillPoints.merge(skillStat.getSkill(), x.value(), Integer::sum);
+                    }
+                });
+            } else if (wynnItemOptional.get() instanceof CraftedGearItem craftedGear) {
+                craftedGear.getIdentifications().forEach(x -> {
+                    if (x.statType() instanceof SkillStatType skillStat) {
+                        craftedSkillPoints.merge(skillStat.getSkill(), x.value(), Integer::sum);
+                    }
+                });
+            } else if (!itemStack.isEmpty()) {
+                WynntilsMod.warn("Skill Point Model failed to parse armour: " + LoreUtils.getStringLore(itemStack));
+            }
+        });
+
+        for (int i : ACCESSORY_SLOTS) {
+            ItemStack itemStack = McUtils.inventory().getItem(i);
+            Optional<WynnItem> wynnItemOptional = Models.Item.getWynnItem(itemStack);
+            if (wynnItemOptional.isEmpty()) continue; // Empty slot
+
+            if (wynnItemOptional.get() instanceof GearItem gear) {
+                gear.getIdentifications().forEach(x -> {
+                    if (x.statType() instanceof SkillStatType skillStat) {
+                        gearSkillPoints.merge(skillStat.getSkill(), x.value(), Integer::sum);
+                    }
+                });
+            } else if (wynnItemOptional.get() instanceof CraftedGearItem craftedGear) {
+                craftedGear.getIdentifications().forEach(x -> {
+                    if (x.statType() instanceof SkillStatType skillStat) {
+                        craftedSkillPoints.merge(skillStat.getSkill(), x.value(), Integer::sum);
+                    }
+                });
+            } else if (!itemStack.isEmpty()
+                    && !itemStack.getHoverName().getString().equals("§7Accessory Slot")) {
+                WynntilsMod.warn("Skill Point Model failed to parse accessory: "
+                        + LoreUtils.getStringLore(McUtils.inventory().getItem(i)));
+            }
```

Comment:
You can extract the highlighted part in both loops into a single method (taking the ItemStack as the arg)